homeassistant/components/calendar/todoist.py

```diff
@@ -217,10 +224,34 @@ def handle_new_task(call):
             date_format = '%Y-%m-%dT%H:%M'
             due_date = datetime.strftime(due_date, date_format)
             item.update(due_date_utc=due_date)
-        # Commit changes
-        api.commit()
+
+        # Commit task
+        result = api.commit()
         _LOGGER.debug("Created Todoist task: %s", call.data[CONTENT])
 
+        # Add default reminder to the task if user has default reminders set
+        user = api.user.get()
+        if user['is_premium']:
+            # Wait for the API to know about the task, otherwise adding
+            # a reminder will fail
+            task = None
+            for _i in range(0, 50):
```

Comment:
This is very bad. We're just hammering the API here. A sleep is also not an option because this method is synchronous. If this was async, you could do `await asyncio.sleep(5)` to wait till API is available.